src/network/apiEndpointCalls.ts

```diff
@@ -0,0 +1,274 @@
+
+  return response;
+}
+
+export async function fetchGameRedditPosts(gameId: number) {
+  const response = await _get<PaginatedRedditPosts>(gameRedditPostsEndpoint(gameId));
+
+  return response;
+}
+
+export async function fetchGameScreenshots(gameId: number) {
+  const response = await _get<PaginatedScreenshots>(gameScreenshotsEndpoint(gameId));
+
+  return response;
+}
+
+export async function fetchGameCollections(gameId: number) {
+  const response = await _get<PaginatedCollectionDetailsResponse>(gameCollectionsEndpoint(gameId));
+
+  return response;
+}
+
+export async function requestAddGameToLibrary(body: AddToLibraryBody) {
+  const response = await _post<AddToLibraryBody, AddToLibraryResponseError>(
+    addToLibraryEndpoint,
+    body,
+  );
+
+  return response;
+}
+
+export async function requestRemoveGameFromLibrary(gameId: number) {
+  const response = await _delete(removeFromLibraryEndpoint(gameId));
+
+  return response;
+}
+
+export async function requestUpdateGameLibrary(gameId: number, body: UpdateGameLibraryStatus) {
+  const response = await _patch<UpdateLibraryResponseSuccess>(
+    updateLibraryStatusEndpoint(gameId),
+    body,
+  );
+
+  return response;
+}
+
+export async function requestAddGameToCollection(
+  collectionId: number,
+  body: AddGameToCollectionBody,
+) {
+  const response = await _post<
+    AddGameToCollectionBody,
+    AddGameToCollectionResponseError,
+    AddGameToCollectionBody
+  >(addGameToCollectionEndpoint(collectionId), body);
+
+  return response;
+}
+
+export async function requestRemoveGameFromCollection(collectionId: number, feedId: number) {
+  const response = await _delete(removeGameFromCollectionEndpoint(collectionId, feedId));
+
+  return response;
+}
+
+const {
+  collectionFeed: collectionFeedEndpoint,
+  addNewCollection: addNewCollectionEndpoint,
+  updateDeleteCollection: updateDeleteCollectionEndpoint,
+} = collectionsEndpoint;
+
+export async function fetchCollectionFeed(collectionId: number) {
+  const response = await _get<PaginatedCollectionFeedsResponse>(
+    collectionFeedEndpoint(collectionId),
+  );
+
+  return response;
+}
+
+export async function requestAddNewCollection(newCollection: AddNewOrUpdateCollectionBody) {
+  const response = await _post<CollectionDetailResponse, AddNewCollectionResponseError>(
+    addNewCollectionEndpoint,
+    newCollection,
+  );
+
+  return response;
+}
+
+export async function requestUpdateCollection(
+  collectionId: number,
+  updatedCollection: Partial<AddNewOrUpdateCollectionBody>,
+) {
+  const response = await _post<CollectionDetailResponse, AddNewCollectionResponseError>(
+    updateDeleteCollectionEndpoint(collectionId),
+    updatedCollection,
+  );
+
+  return response;
+}
+
+export async function requestDeleteCollection(collectionId: number) {
+  const response = await _delete(updateDeleteCollectionEndpoint(collectionId));
+
+  return response;
```

Comment:
instead of doing this you can directly return `await _delete(updateDeleteCollectionEndpoint(collectionId));`